lib/ansible/modules/cloud/azure/azure_rm_batchaccount.py

```diff
@@ -0,0 +1,307 @@
+        - batch_service
+        - user_subscription
+    state:
+        description:
+        - Assert the state of the Batch Account.
+        - Use 'present' to create or update a Batch Account and 'absent' to delete
+            it.
+        default: present
+        choices:
+        - present
+        - absent
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Junyi Yi (@JunyiYi)"
+'''
+
+EXAMPLES = '''
+  - name: Create (or update) Batch Account
+    azure_rm_batchaccount:
+      resource_group: MyResGroup
+      name: "test_object"
+      location: West US
+      auto_storage_account: MyStorageAccountId
+      state: present
+'''
+
+RETURN = '''
+id:
+    description:
+    - The identifier of the Batch Account resource.
+    returned: always
+    type: str
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+from ansible.module_utils.common.dict_transformations import _snake_to_camel
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrest.polling import LROPoller
+    from msrestazure.azure_operation import AzureOperationPoller
+    from msrest.serialization import Model
+    from azure.mgmt.batch import BatchManagementClient
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMBatchAccount(AzureRMModuleBase):
+    """Configuration class for an Azure RM Batch Account resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                required=True,
+                type='str'
+            ),
+            name=dict(
+                required=True,
+                type='str'
+            ),
+            location=dict(
+                required=True,
+                type='str'
+            ),
+            auto_storage_account=dict(
+                type='str'
+            ),
+            key_vault_reference=dict(
+                type='dict',
+                options=dict(
+                    id=dict(
+                        required=True,
+                        type='str'
+                    ),
+                    url=dict(
+                        required=True,
+                        type='str'
+                    )
+                )
+            ),
+            pool_allocation_mode=dict(
+                default='batch_service',
+                type='str',
+                choices=['batch_service', 'user_subscription']
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.batch_account = dict()
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        required_if = [
```

Comment:
seems useless

Reply:
I'm not sure what to put here, so I just left these lines as a placeholder. Let me remove them.